lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -238,145 +274,248 @@ bool Caffe2ModelLoader::hasMultidirectionalBroadcast(
   return false;
 }
 
-llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
-  ArgumentDictionaryTy dict = loadArgumentMap(op);
+llvm::Error Caffe2ModelLoader::loadConv(const caffe2::OperatorDef &op,
+                                        ArgumentDictionaryTy &dict) {
+  const std::string &opName = loadOperatorName(op);
   const std::string &typeName = op.type();
 
-  // Check if operator is supported in parent class, CommonOperatorLoader.
-  bool loadCommonOperatorSuccess;
-  ASSIGN_VALUE_OR_RETURN_ERR(loadCommonOperatorSuccess,
-                             tryLoadCommonOperator(typeName, op, dict));
-  if (loadCommonOperatorSuccess) {
-    return llvm::Error::success();
+  // Load the inputs:
+  std::vector<unsigned_t> strides;
+  ASSIGN_VALUE_OR_RETURN_ERR(strides, getSizeHW(dict, "stride", 1));
+  std::vector<unsigned_t> pads;
+  ASSIGN_VALUE_OR_RETURN_ERR(pads, getPads(dict));
+  std::vector<unsigned_t> kernels;
+  ASSIGN_VALUE_OR_RETURN_ERR(kernels, getSizeHW(dict, "kernel", 0));
+  unsigned_t group = 1;
+  if (dict.count("group")) {
+    ASSIGN_VALUE_OR_RETURN_ERR(group, loadInt(dict["group"]));
+  }
+  std::string order = "NCHW";
+  if (dict.count("order")) {
+    ASSIGN_VALUE_OR_RETURN_ERR(order, loadStr(dict["order"]));
   }
-  const std::string &opName = loadOperatorName(op);
 
-  if (typeName == "Conv" || typeName == "ConvRelu" || typeName == "Int8Conv" ||
-      typeName == "Int8ConvRelu") {
-    // Load the inputs:
-    std::vector<unsigned_t> strides;
-    ASSIGN_VALUE_OR_RETURN_ERR(strides, getSizeHW(dict, "stride", 1));
-    std::vector<unsigned_t> pads;
-    ASSIGN_VALUE_OR_RETURN_ERR(pads, getPads(dict));
-    std::vector<unsigned_t> kernels;
-    ASSIGN_VALUE_OR_RETURN_ERR(kernels, getSizeHW(dict, "kernel", 0));
-    unsigned_t group = 1;
-    if (dict.count("group")) {
-      ASSIGN_VALUE_OR_RETURN_ERR(group, loadInt(dict["group"]));
-    }
-    std::string order = "NCHW";
-    if (dict.count("order")) {
-      ASSIGN_VALUE_OR_RETURN_ERR(order, loadStr(dict["order"]));
-    }
+  NodeValue in;
+  ASSIGN_VALUE_OR_RETURN_ERR(in, getNodeValueByName(op.input(0)));
+
+  Constant *w;
+  ASSIGN_VALUE_OR_RETURN_ERR(w, getConstantByName(op.input(1)));
+
+  // Transpose the weights to the right format. Glow expects to read the
+  // weights in the format CRSK.
+  // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
+  // Caffe2 "Conv" op always stores the weight as CKRS.
+  Tensor wT;
+  w->getPayload().transpose(&wT, NCHW2NHWC);
```

Comment:
So given this is not checking `order` at all, are we only supporting `NCHW` here? If so I think it would make sense to check that `order` is `NCHW` after assigning it above, and returning an error otherwise. Then you could get rid of the logic below checking `order`.

Reply:
This should be matching the original code which had the following comment
```
// Caffe2 "Conv" op always stores the weight as CKRS, while for "Int8Conv",
// and "Int8ConvRelu", the weights always follows the "order" arg.
```
Since Glow expects CRSK and `Conv` and `ConvRelu` always use CKRS this should always transpose them.

Reply:
My understanding based on the previous code is that we the weights are always stored as CKRS regardless of order so we should not need to check the order here (though it's still needed for inputs/outputs)